extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -87,7 +108,7 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.mutateElement(() => {
       createShadowRootWithStyle(
```

Comment:
I just realized we're rendering the tag every time `SHOPPING_DATA` updates.
We need to only render this once.

We need an early return if the tag has already been rendered.
An approach could be to set a private variable to the return value of `createShadowRootWithStyle`.
Then check if it exists already before proceeding with the function.
Something like:
```
  createAndAppendInnerShoppingTagEl_(shoppingData) {
    const tagData = shoppingData[this.element.getAttribute('data-tag-id')];
    if (this.renderedEl_ || !tagData) {
      return;
    }
    this.mutateElement(() => {
      this.renderedEl_ = createShadowRootWithStyle(
        this.element,
        renderShoppingTagTemplate(tagData, (tagData) => this.onClick_(tagData)),
        shoppingTagCSS
      );
    });
  }
```

Or we could have a boolean flag like `this.rendered_` that we set and check. Either way will work.

Reply:
Sounds good, I added the first approach that you suggested.